pkg/apis/kops/cluster.go

```diff
@@ -593,9 +593,6 @@ func (c *Cluster) FillDefaults() error {
 	} else if c.Spec.Networking.AmazonVPC != nil {
 		// OK
 	} else if c.Spec.Networking.Cilium != nil {
-		if c.Spec.Networking.Cilium.Version == "" {
-			c.Spec.Networking.Cilium.Version = CiliumDefaultVersion
```

Comment:
Instead of baking the logic in the templates, this could have picked the default version by doing check on the Kubernetes version. That would allow other Go code to do Cilium version checks. But we don't have any such Cilium version checks, so not a blocker.